<meta charset="UTF-8">
<html>
  <head>
    <link rel="stylesheet" href="src.css"/>
	 <script src="src.js"></script>
    <script src="CommonIssuesData.js"></script>
    <script>
  var issues='{"fileName":"UserLogic.cs","relativePath":"../../../Logic/UserLogic.cs","absolutePath":"D:/DOCUMENTOS/ORT/2023s2/da2/242493-260956-281651/ECommerce/Logic/UserLogic.cs","projectName":"Logic","line":"100","statement":"30","debt":"1h  45min","cov":"N/A","issues":[{"line":"10","ruleId":"ND1900","debt":"5min","severity":"1","expl":"The <i>field _userRepository</i> can be declared as <b>readonly</b>. Doing so explicitly conveys that the field should not be altered after initialization and enforces this constraint."},{"line":"16","ruleId":"ND2207","debt":"20min","severity":"2","expl":"The <i>method CreateUser(…)</i> creates an instance of the <i>class Exception</i>. Throwing such exception compels client code to catch all exceptions, including unknown exceptions that they do not know how to handle."},{"line":"31","ruleId":"ND2207","debt":"20min","severity":"2","expl":"The <i>method DeleteUser(…)</i> creates an instance of the <i>class Exception</i>. Throwing such exception compels client code to catch all exceptions, including unknown exceptions that they do not know how to handle."},{"line":"47","ruleId":"ND2207","debt":"20min","severity":"2","expl":"The <i>method EditUser(…)</i> creates an instance of the <i>class Exception</i>. Throwing such exception compels client code to catch all exceptions, including unknown exceptions that they do not know how to handle."},{"line":"71","ruleId":"ND2207","debt":"20min","severity":"2","expl":"The <i>method GetAllUsers()</i> creates an instance of the <i>class Exception</i>. Throwing such exception compels client code to catch all exceptions, including unknown exceptions that they do not know how to handle."},{"line":"84","ruleId":"ND2207","debt":"20min","severity":"2","expl":"The <i>method GetUser(…)</i> creates an instance of the <i>class Exception</i>. Throwing such exception compels client code to catch all exceptions, including unknown exceptions that they do not know how to handle."}]}';
    </script>
  </head>
  <body>
    <span id="source" fid="5"  style="display:none">UEsDBBQAAAAIAJuacFczuj0eVQIAAEkLAAAKAAAAemlwcGVkLnR4dLVVTW/bMAy9B8h/YHPoHGzwD2jXAVvrFQGGHTps10K1mESAImWSnLQY/N8nyd+2nHZdrUMcynx6FB9JZ5qJDdzIHWHicj7LvJk8prg3TAodf5MbljZ27eL3V8KgWpMUm/073EvNjFRP9Uv7bj4TZId6b60COZ/9mc/Arn32wFkKKSdaw0+Nyr+GC1jVRuFY+nuMYgdisHBpCOE+69iet8YUPPWhUR/cxS4bZIvYLbNlOu4xwRUMqCv/PBgDXCu0V3B/I287/DgpW0PU5zy7ApFxvux69oBuOVz8A/n6F+GMEqdjtLwc+gVI4uSRaRP5ExJbIny5HOKKrCh5BIFH+ELoHf7OUJu6aqKFxwLh9tL0CbZEAwGFG3s2KqRA0lRmwixCUSk0mRLBtMetJPr89eB510Su8dlkNRdphX9NxDsDqWfz6RxEmncEH0h+kIzCDXIso73NrM3oRII78Fk/Vy12SxyQMXBQNyHfpflqZaKtxPjSPTKztZe5WMB7+7A/CypRg5AG0JVPUNZ8Wq2ov+2oVqfV8rdKKDNtrT4U2wpTZAecSrrq+Jf0q4/HSBeoHUB9vW/RVGKP9HoJrWIbKwDnWodVNLKFFODSPj8f0BeDo4sbGx4jxG5NN1fcysPbp0ZOXRauIupiCBX4cGtY1Cdu/3zfuQZbuxfhBpu0v9BVzmsm4SoR2Q4VeeD40d3iE9hC/cy5+6+jN+qq0a9IoEka7mkn0gaNLVTuk6b/NWte76qjgx+PN5xB5Zx73VBxuDBTtUop7CNwxP9U/QSCvaDCy0f+F1BLAQIUABQAAAAIAJuacFczuj0eVQIAAEkLAAAKAAAAAAAAAAAAAAAAAAAAAAB6aXBwZWQudHh0UEsFBgAAAAABAAEAOAAAAH0CAAAAAA==</span>
  </body>
</html>
